forms-flow-web/src/helper/regExp/inputValidater.js

```diff
@@ -0,0 +1,16 @@
+const inputValidater = (searchText) => {
+    let char = '';
+    const a = searchText;
+    if (searchText !== '') {
+        for (var i = 0; i < a?.length; i++) {
+            char = a.charCodeAt(i);
+            if (((char > 32) && (char < 45)) || ((char > 45) && (char < 48))
```

Comment:
Add proper comments .   invalidChars to avoid is this right  /[^A-Za-z0-9]/g; ?

Reply:
Also try proper formatting

Reply:
`const inputValidater = (searchText) => {
  if (!searchText) {
    return false;
  }

  for (let i = 0; i < searchText.length; i++) {
    const char = searchText.charCodeAt(i);
    if (
      (char < 48) ||
      (char > 57 && char < 65) ||
      (char > 90 && char < 97) ||
      (char > 122)
    ) {
      return false;
    }
  }

  return true;
};`

Reply:
also can update the code to a for of loop in this case , try to get the code more readable and cleaner
```
const inputValidater = (searchText) => {
  if (!searchText) {
    return false;
  }

  let valid = true;

  for (const char of searchText) {
    const charCode = char.charCodeAt(0);
    if (
      charCode < 48 ||
      (charCode > 57 && charCode < 65) ||
      (charCode > 90 && charCode < 97) ||
      charCode > 122
    ) {
      valid = false;
      break;
    }
  }

  return valid;
}
```

Reply:
if (!searchText) {
    return false;
  }
adding this lines of code will return false while removing the input using backspace and will lead to the validation error like remove the special charactors in the empty input box. 
updated the code with comments and for loop.